### DISCUSSION TITLE
Non-Modal Table Action Links in Relation Manager

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#3586] [General]
Is there a way to force the table action links in a Relation Manager to act like the table action links in the resource itself - meaning NOT load in a modal but use a real link instead? If I just use this code, the table action links default to loading modals:

```php
public static function table(Table $table): Table
{
    return ShowResource::table($table);
}
```

In order to get this to use real links, I have to do the following:

```php
public static function table(Table $table): Table
{
    return ShowResource::table($table)
        ->actions([
            Tables\Actions\ViewAction::make()
                ->iconButton()
                ->url(fn ($record) => ShowResource::getUrl('view', ['record' => $record])),
            Tables\Actions\EditAction::make()
                ->iconButton()
                ->url(fn ($record) => ShowResource::getUrl('edit', ['record' => $record])),
        ]);
}
```

This isn't necessarily a problem, but wondering if there is a simple flag or something that I could change to avoid re-declaring these actions on the relation manager when they are already declared on the original resource.

**Comment by invaders-xx** [comment]
Try https://filamentphp.com/tricks/relation-manager-edit-record-without-modal

**Reply by danharrin** [reply]
@intrepidws has done this in their example, but wants to use this behaviour by default instead.

The reason why it's not possible atm, is because relation managers have no knowledge that a related resource exists. Also, IMO, the UX is worse using links instead of modals, as owner record context is lost.

**Reply by intrepidws** [reply]
I don't disagree that it's worse in UX most situations - but occasionally it makes more sense IMO.

No worries, just figured I'd see if it was possible.

Thanks @invaders-xx, I hadn't seen that. I eventually ended up on similar functionality but it took me a bit - wish I had seen that first.

**Reply by danharrin** [reply]
It's impressive that you got it first time, exactly correct. Well done. Either way, it's thankfully only 1 line per action to add the url functionality.
